packages/core/directives/map.ts

```diff
@@ -418,7 +411,7 @@ export class AgmMap implements OnChanges, OnInit, OnDestroy {
       rotateControlOptions: this.rotateControlOptions,
       fullscreenControl: this.fullscreenControl,
       fullscreenControlOptions: this.fullscreenControlOptions,
-      mapTypeId: this.mapTypeId,
+      mapTypeId: this.mapTypeId.toLowerCase() as google.maps.MapTypeId,
```

Comment:
no, Google maps is using strings as enum. This **is** the same thing and is a simpler way to convert keys to values until we have the enum key -> value function you're suggesting above.

Reply:
Ok, I just now connected the `toLowerCase` with the conversation above. 

While google _is_ using strings as enum, the @types/googlemaps package is using enum string for mapTypeId, and enum string is not a string. 

AFAIK, enums is the only thing that remains from typescript/types after ts->js compilation, and the only reason to include the `@types/*` libs in bundle. During compilation, a string enum will be converted to an object with string values, which will match "enums" from google maps, so there won't be runtime errors.

The reason I don't like this, is that we are circumventing Typescript's type systems, and the type safety it provides. Sure, TypeScript is nice enough to allow us to easily circumvent it (`as any as *` comes to mind), but 'fidgeting with compile system so the end product runs' feels like an anti-pattern. 

This is why, while your solution *is* more concise, and it is stable in runtime, I still think it's wrong.

Reply:
Just to be clear **I** updated @types/googlemaps, and it uses an `enum` for a string rather than `const enum` because that's what you're supposed to use in DefinitelyTyped. The @types packages **does not** create an enum. The values are coming from the Google Maps JS and they **are not** enums. They are strings.

You are correct, that for a **TypeScript** package the enums are not compiled out, but @types/googlemaps is not adding any code to the compiled output but **instead** defining what types the code should be expecting. There is no conversion as you are suggesting, the compiler just drops the type information and all is good.

I believe the above answers your last paragraph and is why this should be allowed (of course if we **really** would not like to accept the enum keys then this doesn't matter here. I was merely trying to make this behave the same way we've done it elsewhere)

Reply:
Ultimately, I just don't agree to circumventing typescript's type system, even though typescripts type system is more transient and premissive than Java's, for example, and even if it disappears after compilation. A `MapTypeId` is *NOT* a string, nor can it be cast to a string. Casting an enum to a string is *incorrect*, even if it will work out fine after compilation. 

Additionally, I don't think typescript compiler will allow you to do 
```
<agm-map [mapTypeId]="'hybrid'" ...>
```
because 'hybrid' is neither a `MapTypeId`, nor a `keyof typeof MapTypeId`.
Notice that even when you set the default, you are forced to cast it, otherwise compiler complains.

I think that the best solution, is keeping the directive's Input type as 
`'roadmap' | 'hybrid' | 'satellite' | 'terrain'`, and in the manager, converting to MapTypeId.

A second best solution would be 
`google.maps.MapTypeId | keyof typeof google.maps.MapTypeId = 'ROADMAP'`: that is only accept upper case. This will break existing code, but be more aligned with other enums.

If there are other places in the code, where we treat an enum as a string, we should fix it, either now if it was caused by this PR, or eventually.